Rock.Slingshot/BulkImport/BulkImportHelper.cs

```diff
@@ -128,15 +128,15 @@ public string BulkAttendanceImport( List<AttendanceImport> attendanceImports, st
             var attendancesAlreadyExistForeignIdHash = new HashSet<int>( qryAttendancesWithForeignIds.Select( a => a.ForeignId.Value ).ToList() );
 
             // Get list of existing occurrence records that have already been created
-            var existingOccurrences = new HashSet<ImportOccurrence>( new AttendanceOccurrenceService( rockContext ).Queryable()
-                .Select( o => new ImportOccurrence
+            var existingOccurrencesLookup = new AttendanceOccurrenceService( rockContext ).Queryable()
+                .Select( o => new
                 {
                     Id = o.Id,
                     GroupId = o.GroupId,
                     LocationId = o.LocationId,
                     ScheduleId = o.ScheduleId,
                     OccurrenceDate = o.OccurrenceDate
-                } ) );
+                } ).ToDictionary( k => $"{k.GroupId}|{k.LocationId}|{k.ScheduleId}|{k.OccurrenceDate}", v => v.Id );
```

Comment:
@JoshuaMaiorino Nice optimization!

Reply:
@JoshuaMaiorino Oops, somehow this showed up in the pull request, but that was done by a different developer